src/argument_parser.rs

```diff
@@ -3,37 +3,21 @@ use std::{borrow::Cow, path::PathBuf};
 /// Get `output_path`, specified by `-o`
 pub fn get_output_path(args: &[String]) -> crate::Result<&String> {
     args.windows(2)
-        .find_map(|x| {
-            if x[0] == "-o" {
-                return Some(&x[1]);
-            }
-            None
-        })
-        .ok_or_else(|| "(BUG?) `-o` flag not found".to_string().into())
+        .find_map(|x| (x[0] == "-o").then(|| &x[1]))
+        .ok_or_else(|| "(BUG?) `-o` flag not found".into())
 }
 
 /// Get `search_paths`, specified by `-L`
 pub fn get_search_paths(args: &[String]) -> Vec<PathBuf> {
     args.windows(2)
-        .filter_map(|x| {
-            if x[0] == "-L" {
-                log::trace!("new search path: {}", x[1]);
-                return Some(PathBuf::from(&x[1]));
-            }
-            None
-        })
+        .filter_map(|x| (x[0] == "-L").then(|| PathBuf::from(&x[1])))
```

Comment:
Why `find()` in the first case and `filter()` here? I don't get the difference, is it consistent?

Reply:
`find_map` gives you a single item, while `filter_map` gives you a list of items.